### PR TITLE
decompressor bzip2 support

### LINKED ISSUE #309: Add Support for bzip2 decompression in Decompressor
### 🚀 The feature

As of now [Decompressor](https://github.com/pytorch/data/blob/main/torchdata/datapipes/iter/util/decompressor.py) does not have support for .bz2 files. This could be added.

### Motivation, pitch

I am currently working on https://github.com/pytorch/vision/issues/5352 with the [USPS](https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass.html#usps) dataset which uses bzip2. I had to use bz2 directly. Would've liked to use Decompressor.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lezwon**
Fixes #309

### Changes

- Added bzip2 support in decompressor
- Added Bz2FileLoader

### CODE REVIEW

File: torchdata/datapipes/iter/__init__.pyi [hunk 1]
```diff
@@ -154,6 +155,7 @@ __all__ = [
     "ZipArchiveLoader",
     "ZipArchiveReader",
     "Zipper",
+    "Bz2FileLoader",
```
Comment (NivekT) [inline torchdata/datapipes/iter/__init__.pyi:158]: Alphabetical order here too

**Comment by NivekT** [comment]
Hi @lezwon, can you rebase this PR? This is because `__init__.pyi` has been removed by another PR, so I need a rebase before I can merge. Thanks!

**Comment by lezwon** [comment]
@NivekT I see some internal tests have failed. Can you help me out here so I can fix them?

**Comment by NivekT** [comment]
> @NivekT I see some internal tests have failed. Can you help me out here so I can fix them?

Everything is fine. Thanks for double-checking.

**Pull request closed without merging**